pkg/webhook/csidriver/csidriver.go

```diff
@@ -81,6 +101,72 @@ func (s *SharedResourcesCSIDriverWebhook) authorizePod(request admissionctl.Requ
 	return ret
 }
 
+func (s *SharedResourcesCSIDriverWebhook) authorizedsharedSecret(request admissionctl.Request) admissionctl.Response {
+	klog.V(2).Info("admitting shared secret with SharedResourceCSIDiver")
+	var (
+		ret           admissionctl.Response
+		allowResource bool
+	)
+
+	SharedSecret, err := s.renderSharedSecret(request)
+	if err != nil {
+		klog.Error(err, "Couldn't render a shared secret from the incoming request")
+		return admissionctl.Errored(http.StatusBadRequest, err)
+	}
+
+	if strings.HasPrefix(SharedSecret.ObjectMeta.Name, "openshift-") {
+		for _, value := range config.LoadedConfig.AllowedSharedResources {
+			if strings.Contains(value, SharedSecret.ObjectMeta.Name) {
+				allowResource = true
+				break
+			}
+		}
+		if !allowResource {
+			admissionctl.Denied("Shared secrets are not allowed to be created with the reserved prefix \"openshift-\" unless its name and Reference key are found in a well known list of OCP pre-populated Shared Resource")
```

Comment:
This is probably a bit long and verbose, I think it can be cut down to a simplified message.